### PR TITLE
Implement a way to define listeners options

### PR DESCRIPTION
**Pull request by WebReflection**
some event cannot be properly set (i.e. `{once: true}` or `{passive: true}` or others).

the goal here is not to polyfill these events, since there are other libraries for this such as [dom4](https://github.com/WebReflection/dom4), rather provide an option that scales out of the box.

The `on${event.type}Options` name can be either absent or `false`, `true`, or any object containing options.

```js
wickedElements.define('.click-once', {
  onclickOptions: {once: true},
  onclick() { alert('clicked once'); }
});
```

### CODE REVIEW

**Comment by grommett** [comment]
Another good idea! 💯 

How does this work for custom events, though? I'm using them pretty extensively. At the moment I don't need any of the listener options, though...

```js
wickedElements.define('.element-with-custom-event', {
  oncustomEventOptions: {once: true}, // <--- ?
  oncustomEvent() { alert('custom event fired once'); }
});
```

**Comment by WebReflection** [comment]
> How does this work for custom events, though?

the same, if you need a custom event to be fired once, pass `{once: true}` and be done with it.
events are always handled natively, and options are a native feature too, even if an event is custom.

> At the moment I don't need any of the listener options, though...

That means there's nothing you should do, the `on...Options` is optional, it defaults to `false` as it's always been.

Does this answer your concerns?

File: cjs/index.js [hunk 1]
```diff
@@ -86,25 +84,32 @@ function addIfNeeded(component, key, value) {
...
     var i = 0;
     var length = keys.length;
     while (i < length) {
       var key = keys[i++];
-      if (key.slice(0, 2) === 'on')
-        events.push(key.slice(2));
+      if (key.slice(0, 2) === 'on' && key.slice(-7) !== 'Options')
+        events.push({
+          type: key.slice(2),
+          options: proto[key + 'Options'] || false
```
Comment (grommett) [inline cjs/index.js:111]: This answers the custom event question :)

**Comment by grommett** [comment]
> Does this answer your concerns?

Absolutely, thanks!

**Pull request merged**